Assignment 3.Rmd

```diff
@@ -117,6 +153,112 @@ Once you have done this, also [look up](http://igraph.org/r/) how to generate th
 * Betweeness centrality and dregree centrality. **Who is the most central person in the network according to these two metrics? Write a sentence or two that describes your interpretation of these metrics**
 
 * Color the nodes according to interest. Are there any clusters of interest that correspond to clusters in the network? Write a sentence or two describing your interpetation.
+```{r}
+library(tidyr)
+library(dplyr)
+library(stringr)
+library(igraph)
+
+
+#Input data
+C1 <- read.csv("hudk4050-classes.csv", stringsAsFactors = FALSE, header = TRUE)
+#Copy to play with data?
+C2 <- C1
+```
+
+#Data Tidying
+```{r}
+#Make header first row
+colnames(C2) <- C2[1,]
+#Remove unwanted rows(can choose other?)
+C2 <- slice(C2,3:49)
+#Remove last column
+C2 <- select(C2, 1:8)
+#Merge name columns
+C2 <- unite(C2, "name", 'First Name', 'Last Name', sep = " ")
+#Remove unpredicted characters from names
+C2$name <- str_replace(C2$name, "`", "")
+#Capitalize only first letter
+C2$name <- str_to_title(C2$name)
+#Class letter capital
+C2 <- C2 %>% mutate_at(2:7, list(toupper))
+#Remove whiteness
+C2 <- C2 %>% mutate_at(2:7, str_replace_all, " ", "")
+```
+#Data Restructuring
+```{r}
+C3 <- C2 %>% gather(label, class, 2:7, na.rm = TRUE, convert = FALSE) %>% select(name, class)
+
+#Create a new variable containing 1s that will not become the counts
+C3$count <- 1
+
+#Remove blank classes
+C3 <- filter(C3, class != "")
+
+#Remove duplicates
+C3 <- unique(C3)
+
+#Spread
+C3 <- spread(C3, class, count)
+
+#Make row names student names
+rownames(C3) <- C3$name
+
+
+
+#Shortest:
+#C3[is.na(C3)] <- 0
+
+#Remove names column and HUDK4050
+C3 <- select(C3, -name, -HUDK4050)
+
+#Cheat way:
+C3 <- ifelse(is.na(C3), 0, 1)
+
+
+
+```
+
+#Matrix operations
+```{r}
+C4 <- as.matrix(C3)
+
+#Create person to person matrix
+C4 <- C4 %*% t(C4)
+```
+
+#Graphing
+```{r}
+g <- graph.adjacency(C4, mode="undirected", diag = FALSE)
+plot(g, layout=layout.fruchterman.reingold,
+     vertex.size = 4, 
+     #degree(g)*0.7,
+     vertex.label.cex=0.8,
+     vertex.label.color="blue",
+     vertex.color="black")
+```
+
+#Centrality
+```{r}
+#Calculate the degree centrality of the nodes, showing who has the most connections
+sort(degree(g), decreasing = TRUE)
+
+#Calculate the betweeness centrality
+sort(betweenness(g), decreasing = TRUE)
+
+```
+#Clearly Yifei Zhang has the highest 'betweeness centrality'. 
```

Comment:
#Clearly 'Yifei Zhang' has the highest 'betweeness centrality'. Yifei has the degree of "24", which is the second highest in the class. And she has the betweenness(g) of 260.6143603, which is the highest in the class. Therefore, she is the most central person in the course while discussing the p-p network (courses).